Rework/improve the vector store documentation (#52874)
dotnet/docs: docs/ai/vector-stores/define-your-data-model.md

#### [Record definition](#tab/key-record-definition)

Use the <xref:Microsoft.Extensions.VectorData.VectorStoreKeyProperty> class to indicate that your property is the key of the record.

:::code language="csharp" source="./snippets/conceptual/schema-with-record-definition.cs" range="12":::

The following table shows the configuration settings for `VectorStoreKeyProperty`.

| Parameter     | Required | Description          |
|---------------|:--------:|----------------------|
| `Name`        | Yes      | The name of the property on the data model. Used by the mapper to automatically map between the storage schema and data model and for creating indexes. |
| `Type`        | No       | The type of the property on the data model. Used by the mapper to automatically map between the storage schema and data model and for creating indexes. |
| `StorageName` | No       | Can be used to supply an alternative name for the property in the database. This parameter is not supported by all providers, for example, where alternatives like `JsonPropertyNameAttribute` are supported. |

---

### Data property

> Data properties hold general-purpose content such as text, tags, or other metadata that are retrieved when searching for records, and can optionally also be indexed for filtering.

Author: Makes sense! Will change.

| `DistanceFunction`   | No       | The type of function to use when doing vector comparison during vector search over this vector. Default varies by vector store type.                                                                         |
| `StorageName`        | No       | Can be used to supply an alternative name for the property in the database. This parameter is not supported by all providers, for example, where alternatives like `JsonPropertyNameAttribute` is supported. |
| `EmbeddingGenerator` | No       | Allows specifying a `Microsoft.Extensions.AI.IEmbeddingGenerator` instance to use for generating embeddings automatically for the decorated property.                                                        |

---

Common index kinds and distance function types are supplied as static values on the <xref:Microsoft.Extensions.VectorData.IndexKind> and <xref:Microsoft.Extensions.VectorData.DistanceFunction> classes. Individual vector store implementations might also use their own index kinds and distance functions, where the database supports unusual types.

## Vector properties and embedding generation

> Vector databases are all about storing *embeddings* - or numerical representations of your data - which are generated by an embedding model. When storing or searching data, embedding generation must first be performed to convert the searchable data to such embeddings. MEVD provides two approaches to doing this:

Reviewer: ```suggestion
Vector databases are all about storing *embeddings* - or numerical representations of your data - which are generated by an embedding model. When storing or searching data, embedding generation must be performed first to convert the searchable data to such embeddings. MEVD provides two approaches to embedding generation: manual and automatic.
```
